### PR TITLE
Changing bqueryd for allowing downloading from Azure or AWS

### PR DESCRIPTION
**Pull request by dkailthya**

### CODE REVIEW

File: bqueryd/node.py [hunk 1]
```diff
@@ -26,18 +27,19 @@ def main(argv=sys.argv):
...
     elif 'movebcolz' in argv:
         bqueryd.MoveBcolzNode(redis_url=redis_url, loglevel=loglevel).go()
     else:
         if len(argv) > 1 and argv[1].startswith('tcp:'):
-            rpc = bqueryd.RPC(address=argv[1], redis_url=redis_url, loglevel=loglevel)
+            rpc = bqueryd.RPC(address=argv[1], redis_url=redis_url, loglevel=loglevel,
+                              azure_conn_string=azure_conn_string)
         else:
-            rpc = bqueryd.RPC(redis_url=redis_url, loglevel=loglevel)
+            rpc = bqueryd.RPC(redis_url=redis_url, loglevel=loglevel, azure_conn_string=azure_conn_string)
```
Comment (dkailthya) [inline bqueryd/node.py]: Why do we need to pass the azure_conn_string here?

File: bqueryd/rpc.py [hunk 2]
```diff
@@ -136,7 +137,7 @@ def _rpc(*args, **kwargs):
 
     def distribute(self, filenames, bucket):
```
Comment (dkailthya) [inline bqueryd/rpc.py]: Can we check if distribute() is used anywhere in our code? I don't think so - if it's dead code, we should remove it.

File: bqueryd/rpc.py [hunk 3]
```diff
@@ -150,16 +151,20 @@ def distribute(self, filenames, bucket):
             # Try to compress the whole bcolz direcory into a single zipfile
             tmpzip_filename, signature = bqueryd.util.zip_to_file(filepath, bqueryd.INCOMING)
 
-            session = boto3.Session()
-            credentials = session.get_credentials()
-            if not credentials:
-                raise ValueError('Missing S3 credentials')
+            if self.azure_conn_string is None:  # AWS
```
Comment (dkailthya) [inline bqueryd/rpc.py]: use if not x: instead of if x is None:

File: bqueryd/worker.py [hunk 4]
```diff
@@ -506,6 +515,70 @@ def _get_s3_conn(self):
         s3_conn = boto3.resource('s3')
         return access_key, secret_key, s3_conn
 
+    def download_file_azure(self, ticket, fileurl)
+        tmp = fileurl[8:].split('/')
```
Comment (dkailthya) [inline bqueryd/worker.py]: use tmp = flleurl.replace('azure://',  '') instead of fileurl[8:]

File: bqueryd/worker.py [hunk 5]
```diff
@@ -506,6 +515,70 @@ def _get_s3_conn(self):
...
+        else:
+            self.logger.info(
+                "Downloading ticket [%s], container name [%s], blob name [%s]" % (ticket, container_name, blob_name))
+
+            blob_client = BlobClient.from_connection_string(conn_str=self.azure_conn_string, container=container_name,
+                                                            blob=blob_name)
+            properties = blob_client.get_blob_propèrties()
+            size = properties.size
+
+            key = 'azure://{}/{}'.format(container_name, blob_name)
```
Comment (dkailthya) [inline bqueryd/worker.py]: This doesn't work - We'll need to use blob_client.download methods instead of smart_open.

File: bqueryd/node.py (hunk 1 above)
Comment (albertosml) [reply bqueryd/node.py]: Due to it's needed for `distribute` method in RPC class, but I have seen that this method isn't called by anything, so we can remove this pass and `distribute` method.

File: bqueryd/rpc.py (hunk 2 above)
Comment (albertosml) [reply bqueryd/rpc.py]: Same as above.

File: bqueryd/worker.py (hunk 4 above)
Comment (albertosml) [reply bqueryd/worker.py]: Done

File: bqueryd/rpc.py (hunk 3 above)
Comment (albertosml) [reply bqueryd/rpc.py]: Done

Comment (albertosml) [reply bqueryd/rpc.py]: Done

**Pull request closed without merging**